### ISSUE TITLE
Customizable menu

### ISSUE THREAD

**Issue by vinayvenu** [avniproject/avni-client#634]
### Story
As a user of the Avni App, I need to have configurable menu items that I can use to configure stuff like
- Report links
- Documentation links
- Youtube videos
- Whatsapp links (such as for a helpdesk

### Feature definition
- There is an option on the App Designer to define menu items. 
- Menu items can be a Link or a Document (from our documentation)
- An item has an icon, text, type, group and link
- Groups of menu items currently available are - Functionality, Sync, User and Support. Menu items can be added to any of these groups, and will be shown at the end of that group
- The link will be a string template to which the following variables will be available - UserInfo(https://github.com/avniproject/avni-models/blob/master/src/UserInfo.js)
eg: `https://google.com/${user.uuid}`

### In scope
- Break down of existing menu items into Functionality, Sync, User and Support
- Regroup of existing menu items in the above format (Video List, Dashboards are for functionality, Sync related items for sync, Change password and Logout for User and Provide Feedback for Support)
- Ability to configure a Whatsapp link template as a menu item (Whatsapp links are regular links 
- Template has a "user" attribute that can be used

### Out of scope
- Youtube videos
- Documentation
- List of videos

### Tech tasks
- Menu option in App Designer
- Sync of menu items
- Display of menu items
- Using templates to form the url
- I18n of Menu item names

**Comment by petmongrels** [comment]
Functionality = [Dashboards, Video Library, News, Beneficiary Mode],
Sync = [Entity Sync Status, Setup Fast Sync],
User = [Change Password, Logout],
Support = [Feedback, Upload Database]

**Comment by petmongrels** [comment]
Documentation for how to use it.

Define the rule.
https://avni.readme.io/docs/writing-rules#12-hyperlink-menu-item-rule

**Comment by arjunk** [comment]
We should think of following functionality part of the "Menu" feature.  :
- Canned and Custom menu items
- When a new implementation is setup, a pre-designed menu shows up. 
- Pre-designed menu can be changed by the implementer including removing the canned menu items
- Implementer shoud be able to pick and choose which menu items are to be shown on home page vs More section
- Question / Suggestion : Rather than groups should we use the Section terminology. In offline dashboards we use 'Section' for grouping. There could be pros and cons of using different terms vs same from the perspective of implementer mind map and vocabulary.
- At the moment menu is only applicable for android app but we should include DEA also. 

Basically a holistic thinking of the Menu feature. If it does not mean rework later we do it part of the other stories later or otherwise we take these into consideration now.

@vinayvenu @petmongrels

**Comment by petmongrels** [comment]
To Do
- Document how implementers will choose icon name and Groups
-
